src/prompt.c

```diff
@@ -0,0 +1,67 @@
+#include "prompt.h"
+#include "stdio.h"
+
+t_prompt 	*prompt_create(const char *_username, const char *_hostname)
+{
+	t_prompt *prompt;
+
+	prompt = ft_calloc(sizeof(t_prompt), 1);
+	if (!prompt)
+		return (NULL);
+	prompt->username = ft_strdup(_username);
+	prompt->hostname = ft_strdup(_hostname);
+	prompt->buffer = ft_calloc(sizeof(char), BUFFER_SIZE);
+	if (!prompt->buffer || !prompt->username || !prompt->hostname)
+	{
+		prompt_destroy(prompt);
+		return (NULL);
+	}
+	prompt->current_dir = directory_get_current_dir();
+	prompt->error_code = 0;
+	prompt_init_buffer(prompt);
+	return (prompt);
+}
+
+void	prompt_init_buffer(t_prompt *_prompt)
+{
+	ft_strlcat(_prompt->buffer, _prompt->username, ft_strlen(_prompt->username));
+	ft_strlcat(_prompt->buffer, "@", 1);
+	ft_strlcat(_prompt->buffer, _prompt->hostname, ft_strlen(_prompt->hostname));
+	ft_strlcat(_prompt->buffer, ": [", 1);
+	ft_strlcat(_prompt->buffer, _prompt->current_dir);
+	ft_strlcat(_prompt->buffer, "] %");
```

Comment:
using this method, how do you know the index of where everything is in the buffer? 
use the indexes that point to the beginning of error_code, user, path, etc or use snprintf() to build the whole string